renku/api/repository.py

```diff
@@ -251,37 +254,44 @@ def subclients(self, parent_commit):
                 Submodule.iter_items(self.repo, parent_commit=parent_commit)
             ]
         except (RuntimeError, ValueError):
-            # There are no submodules assiciated with the given commit.
+            # There are no submodules associated with the given commit.
             submodules = []
 
-        return self._subclients.setdefault(
-            parent_commit, {
-                submodule: self.__class__(
-                    path=(self.path / submodule.path).resolve(),
+        subclients = {}
+        for submodule in submodules:
+            subpath = (self.path / submodule.path).resolve()
+            is_renku = subpath / Path(self.renku_home)
+
+            if subpath.exists() and is_renku.exists():
+                subclients[submodule] = self.__class__(
+                    path=subpath,
                     parent=(self, submodule),
                 )
-                for submodule in submodules
-            }
-        )
+
+        return subclients
 
     def resolve_in_submodules(self, commit, path):
         """Resolve filename in submodules."""
         original_path = self.path / path
-        if original_path.is_symlink() or str(path
-                                             ).startswith('.renku/vendors'):
+        in_vendor = str(path).startswith('.renku/vendors')
+
+        if original_path.is_symlink() or in_vendor:
             original_path = original_path.resolve()
+
             for submodule, subclient in self.subclients(commit).items():
-                try:
-                    subpath = original_path.relative_to(subclient.path)
-                    return (
-                        subclient,
-                        subclient.find_previous_commit(
-                            subpath, revision=submodule.hexsha
-                        ),
-                        subpath,
-                    )
-                except ValueError:
-                    pass
+                if (Path(submodule.path) / Path('.git')).exists():
```

Comment:
doesn't a submodule by definition have a `.git` directory?

Reply:
There is no guarantee that subclients will resolve to the git repo. For example, you can have a case where submodule is listed in `.gitmodules` and it's pointing to a raw directory. In that case, code explodes here instead of giving user a nice message.

Reply:
Then we should maybe try to do a better check - is this submodule actually listed in `.gitmodules`, if yes, can we re-initialize it

Reply:
I suppose we can leave that for another time and just leave this code as-is for now